xgi/linalg/laplacian_matrix.py

```diff
@@ -221,16 +228,39 @@ def normalized_hypergraph_laplacian(H, sparse=True, index=False):
             "Every node must be a member of an edge to avoid divide by zero error!"
         )
 
-    D = degree_matrix(H)
-    A, rowdict = clique_motif_matrix(H, sparse=sparse, index=True)
+    (
+        incidence,
+        rowdict,
+        _,  # Discard edge name-index mapping
+    ) = incidence_matrix(H, sparse=sparse, index=True)
+
+    Dv = degree_matrix(H)
+    De = np.sum(incidence, axis=0)
 
     if sparse:
-        Dinvsqrt = csr_array(diags(np.power(D, -0.5)))
+        Dv_invsqrt = csr_array(diags(np.power(Dv, -0.5)))
+        De_inv = diags(1 / De)
+
+        if weighted:
+            W = diags([H.edges[edge_idx].get("weight", 1) for edge_idx in H.edges])
+        else:
+            W = identity(H.num_edges)
+
         eye = csr_array((H.num_nodes, H.num_nodes))
         eye.setdiag(1)
```

Comment:
```suggestion
        eye = eye_array(H.num_nodes, H.num_nodes), format="csr")
```
Just realised this does the same in one line